### PR TITLE
[FLINK-12939][docs-zh] Translate "Apache Kafka Connector" page into C…

### PR DESCRIPTION
**Pull request by highfei2011**
## What is the purpose of the change

*Translate "Apache Kafka Connector" page into Chinese*

## Brief change log

  - *I modified the kafka.zh.md file .*

## Verifying this change

*I have verified the changes by starting the build script in preview mode.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): ( no )
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no )
  - The serializers: ( no )
  - The runtime per-record code paths (performance sensitive): ( no )
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: ( no )
  - The S3 file system connector: ( no )

## Documentation

  - Does this pull request introduce a new feature? (  no )
  - If yes, how is the feature documented? ( not documented )

@wuchong  @TisonKun 
Please Take A Look .

### CODE REVIEW

File: docs/dev/connectors/kafka.zh.md [hunk 1]
```diff
@@ -150,44 +139,40 @@ To use the universal Kafka connector add a dependency to it:
...
-1. The topic name / list of topic names
-2. A DeserializationSchema / KafkaDeserializationSchema for deserializing the data from Kafka
-3. Properties for the Kafka consumer.
-  The following properties are required:
-  - "bootstrap.servers" (comma separated list of Kafka brokers)
-  - "zookeeper.connect" (comma separated list of Zookeeper servers) (**only required for Kafka 0.8**)
-  - "group.id" the id of the consumer group
+1. Topic 名称或者 Topics 的名称列表
+2. 用于反序列化 Kafka 数据的 DeserializationSchema 或者 KafkaDeserializationSchema
+3. Kafka consume 的属性。需要以下属性：
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
3. Kafka 消费者的属性。需要以下属性：
```

File: docs/dev/connectors/kafka.zh.md [hunk 2]
```diff
@@ -150,44 +139,40 @@ To use the universal Kafka connector add a dependency to it:
...
+  - "bootstrap.servers" （以逗号分隔的 Kafka broker 列表）
+  - "zookeeper.connect" （以逗号分割的 Zookeeper servers 列表) (**仅限 Kafka 0.8 使用**)
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
  - "zookeeper.connect" （以逗号分割的 Zookeeper servers 列表) (**仅 Kafka 0.8 需要**)
```

File: docs/dev/connectors/kafka.zh.md [hunk 3]
```diff
@@ -150,44 +139,40 @@ To use the universal Kafka connector add a dependency to it:
...
+  - "group.id" 消费组 ID
 
-Example:
+例：
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
示例：
```

File: docs/dev/connectors/kafka.zh.md [hunk 4]
```diff
@@ -150,44 +139,40 @@ To use the universal Kafka connector add a dependency to it:
...
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
 Properties properties = new Properties();
 properties.setProperty("bootstrap.servers", "localhost:9092");
-// only required for Kafka 0.8
+// 仅限 Kafka 0.8 使用
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
// 仅 Kafka 0.8 需要
```

File: docs/dev/connectors/kafka.zh.md [hunk 5]
```diff
@@ -150,44 +139,40 @@ To use the universal Kafka connector add a dependency to it:
...
-	.addSource(new FlinkKafkaConsumer08<>("topic", new SimpleStringSchema(), properties));
+  .addSource(new FlinkKafkaConsumer08<>("topic", new SimpleStringSchema(), properties));
 {% endhighlight %}
 </div>
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
 val properties = new Properties()
 properties.setProperty("bootstrap.servers", "localhost:9092")
-// only required for Kafka 0.8
+// 仅限 Kafka 0.8 使用
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
// 仅 Kafka 0.8 需要
```

File: docs/dev/connectors/kafka.zh.md [hunk 6]
```diff
@@ -197,41 +182,31 @@ stream = env
...
-The Flink Kafka Consumer needs to know how to turn the binary data in Kafka into Java/Scala objects. The
-`DeserializationSchema` allows users to specify such a schema. The `T deserialize(byte[] message)`
-method gets called for each Kafka message, passing the value from Kafka.
+Flink Kafka Consumer 需要知道如何将 Kafka 中的二进制数据转换为 Java 或者 Scala 对象。 `DeserializationSchema` 允许用户指定这样的 schema ，为每条 Kafka 消息调用 `T deserialize(byte[] message)` 方法，从 Kafka 中传递值。
 
-It is usually helpful to start from the `AbstractDeserializationSchema`, which takes care of describing the
-produced Java/Scala type to Flink's type system. Users that implement a vanilla `DeserializationSchema` need
-to implement the `getProducedType(...)` method themselves.
+从 `AbstractDeserializationSchema` 开始通常很有帮助，它负责将生成的 Java 或 Scala 类型描述为 Flink 的类型系统。
+实现带有 vanilla `DeserializationSchema` 的用户需要自己实现 `getProducedType（...）` 方法。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
用户如果要自己去实现一个`DeserializationSchema`，需要自己去实现 `getProducedType(...)`方法。
```

File: docs/dev/connectors/kafka.zh.md [hunk 7]
```diff
@@ -197,41 +182,31 @@ stream = env
...
-the following deserialize method `T deserialize(ConsumerRecord<byte[], byte[]> record)`.
+为了访问 Kafka 消息的 key 、 value 和元数据， `KafkaDeserializationSchema` 具有以下反序列化方法 `T deserialize(ConsumerRecord<byte[], byte[]> record)` 。
 
-For convenience, Flink provides the following schemas:
+为了方便使用， Flink 提供了以下几种 schemas ：
 
-1. `TypeInformationSerializationSchema` (and `TypeInformationKeyValueSerializationSchema`) which creates
-    a schema based on a Flink's `TypeInformation`. This is useful if the data is both written and read by Flink.
-    This schema is a performant Flink-specific alternative to other generic serialization approaches.
+1. `TypeInformationSerializationSchema` （和 `TypeInformationKeyValueSerializationSchema`) 基于 Flink 的 `TypeInformation` 创建 `schema` 。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
1. `TypeInformationSerializationSchema` （和 `TypeInformationKeyValueSerializationSchema`) 
是一个基于 Flink 的`TypeInformation` 创建的 schema。
```

File: docs/dev/connectors/kafka.zh.md [hunk 8]
```diff
@@ -197,41 +182,31 @@ stream = env
...
+    如果 Flink 既负责数据的读也负责写，那么这将是非常有用的。此 schema 是其他通用序列化方法的高性能 Flink 替代方案。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
    如果 Flink 既负责数据的读也负责写，那么这将是非常有用的。此 schema 是其他通用序列化方法的高性能 Flink 替代方案。
```
```suggestion
    如果该数据的读和写都发生在 Flink 中，那么这将是非常有用的。此 schema 是其他通用序列化方法的高性能 Flink 替代方案。
```

File: docs/dev/connectors/kafka.zh.md [hunk 9]
```diff
@@ -197,41 +182,31 @@ stream = env
...
 
-2. `JsonDeserializationSchema` (and `JSONKeyValueDeserializationSchema`) which turns the serialized JSON
-    into an ObjectNode object, from which fields can be accessed using `objectNode.get("field").as(Int/String/...)()`.
-    The KeyValue objectNode contains a "key" and "value" field which contain all fields, as well as
-    an optional "metadata" field that exposes the offset/partition/topic for this message.
+2. `JsonDeserializationSchema` （和 `JSONKeyValueDeserializationSchema`） 将序列化的 JSON 转化为 ObjectNode 对象，可以使用 `objectNode.get("field").as(Int/String/...)()` 来访问某个字段。
+    KeyValue objectNode 包含一个含所有字段的 key 和 values 字段，以及公开此消息的 offset 或 partition 或 topic 的可选“元数据”字段。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
    KeyValue objectNode 包含一个含所有字段的 key 和 values 字段，以及一个可选的"metadata"字段，可以访问到消息的 offset、partition、topic 等信息。
```

这里 "metadata" 是需要作为 key 去访问的，所以不建议翻译。

File: docs/dev/connectors/kafka.zh.md [hunk 10]
```diff
@@ -254,33 +229,24 @@ For convenience, Flink provides the following schemas:
...
-When encountering a corrupted message that cannot be deserialized for any reason, there
-are two options - either throwing an exception from the `deserialize(...)` method
-which will cause the job to fail and be restarted, or returning `null` to allow
-the Flink Kafka consumer to silently skip the corrupted message. Note that
-due to the consumer's fault tolerance (see below sections for more details),
-failing the job on the corrupted message will let the consumer attempt
-to deserialize the message again. Therefore, if deserialization still fails, the
-consumer will fall into a non-stop restart and fail loop on that corrupted
-message.
+当遇到因一些原因而无法反序列化的损坏消息时，这里有两个选项 - 从 `deserialize（...）` 方法抛出异常会导致作业失败并重新启动，或返回 `null` ，以允许 Flink Kafka 消费者悄悄地跳过损坏的消息。请注意，由于 Consumer 的容错能力（请参阅下面的部分以获取更多详细信息），在损坏的消息上失败作业将使 consumer 尝试再次反序列化消息。因此，如果反序列化仍然失败，则 consumer 将在该损坏的消息上进入不间断重启和失败的循环。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]: How about to translate "corrupted message" as"脏数据"？ 
IMO, "损坏的消息" is not native.

File: docs/dev/connectors/kafka.zh.md [hunk 11]
```diff
@@ -342,74 +300,51 @@ myConsumer.setStartFromSpecificOffsets(specificStartOffsets)
...
+请注意：当 Job 从故障中自动恢复或使用 savepoint 手动恢复时，这些起始位置配置方法不会影响消费的起始位置。在恢复时，每个 Kafka 分区的起始位置由存储在 savepoint 或 checkpoint 中的 offsets 确定（有关 checkpointing 的信息，请参阅下一节，以便为 consumer 启用容错功能）。
 
-### Kafka Consumers and Fault Tolerance
+### Kafka Consumers 和容错
 
-With Flink's checkpointing enabled, the Flink Kafka Consumer will consume records from a topic and periodically checkpoint all
-its Kafka offsets, together with the state of other operations, in a consistent manner. In case of a job failure, Flink will restore
-the streaming program to the state of the latest checkpoint and re-consume the records from Kafka, starting from the offsets that were
-stored in the checkpoint.
+伴随着启用 Flink 的 checkpointing 后，Flink Kafka Consumer 将使用 topic 中的记录，并以一致的方式定期检查其所有 Kafka offsets 和其他算子的状态。``如果 Job 失败， Flink 会将流式程序恢复到最新 checkpoint 的状态，并从存储在 checkpoint 中的 offsets 开始重新消费 Kafka 中的消息。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
伴随着启用 Flink 的 checkpointing 后，Flink Kafka Consumer 将使用 topic 中的记录，并以一致的方式定期检查其所有 Kafka offsets 和其他算子的状态。如果 Job 失败， Flink 会将流式程序恢复到最新 checkpoint 的状态，并从存储在 checkpoint 中的 offsets 开始重新消费 Kafka 中的消息。
```

File: docs/dev/connectors/kafka.zh.md [hunk 12]
```diff
@@ -342,74 +300,51 @@ myConsumer.setStartFromSpecificOffsets(specificStartOffsets)
...
-env.enableCheckpointing(5000) // checkpoint every 5000 msecs
+env.enableCheckpointing(5000) // 每隔 5000 毫秒 执行一次 checkpoint
 {% endhighlight %}
 </div>
 </div>
 
-Also note that Flink can only restart the topology if enough processing slots are available to restart the topology.
-So if the topology fails due to loss of a TaskManager, there must still be enough slots available afterwards.
-Flink on YARN supports automatic restart of lost YARN containers.
+另请注意，如果有足够的处理 solts 是可用于重新启动拓扑的，那么 Flink 只能重新启动拓扑计划。因此，如果拓扑由于丢失了 TaskManager 而失败，那么之后必须要一直有足够可用的 solts 。 Flink on YARN 支持自动重启丢失的 YARN 容器。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
另请注意，Flink 只有在有足够的可用 solts 情况下才重新启动拓扑。因此，如果拓扑由于丢失了 TaskManager 而失败，那么之后必须要有足够可用的 solts 。 Flink on YARN 支持自动重启丢失的 YARN 容器。
```

File: docs/dev/connectors/kafka.zh.md [hunk 13]
```diff
@@ -448,61 +383,36 @@ val stream = env.addSource(myConsumer)
...
+
+ - *禁用 Checkpointing ：* 如果禁用了 checkpointing ，则 Flink Kafka Consumer 依赖于内部使用的 Kafka client 自动定期 offset 提交功能。
+ 因此，要禁用或启用 offset 的提交，只需将 `enable.auto.commit`（或 Kafka 0.8 的 `auto.commit.enable` ）或者 `auto.commit.interval.ms` 的Key 值设置为提供的 `Properties` 配置中的适当值。
+
+ - *启用 Checkpointing ：* 如果启用了 checkpointing ，那么当 checkpointing 完成时， Flink Kafka Consumer 将提交的 offset 存储在 checkpoints 状态中。
+ 这确保 Kafka broker 中提交的 offset 与 checkpoints 状态中的 offset 一致。
+ 用户可以通过调用 consumer 上的 `setCommitOffsetsOnCheckpoints(boolean)` 方法来禁用或启用 offset 的提交(默认情况下，这个值是 true )。
+ 注意，在这个场景中，完全忽略 `Properties` 中的自动定期 offset 提交设置。
+
+### Kafka Consumers 和 Timestamp extractor 或者 Watermark Emission
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
### Kafka Consumers 和时间戳抽取以及 Watermark 发送
```

File: docs/dev/connectors/kafka.zh.md [hunk 14]
```diff
@@ -448,61 +383,36 @@ val stream = env.addSource(myConsumer)
...
+
+在许多场景中，记录的时间戳(显式或隐式)嵌入到记录本身中。此外，用户可能希望定期或以不规则的方式 Watermark Emission ，例如基于 Kafka 流中包含当前事件时间水位线的特殊记录。对于这些情况， Flink Kafka Consumer 允许指定 `AssignerWithPeriodicWatermarks` 或 `AssignerWithPunctuatedWatermarks`。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
在许多场景中，记录的时间戳(显式或隐式)嵌入到记录本身中。此外，用户可能希望定期或以不规则的方式发送 watermark ，例如基于 Kafka 流中包含当前事件时间水位线的特殊记录。对于这些情况， Flink Kafka Consumer 允许指定 `AssignerWithPeriodicWatermarks` 或 `AssignerWithPunctuatedWatermarks`。
```

File: docs/dev/connectors/kafka.zh.md [hunk 15]
```diff
@@ -448,61 +383,36 @@ val stream = env.addSource(myConsumer)
...
+
+你可以按照[此处]({{ site.baseurl }}/dev/event_timestamps_watermarks.html)的说明指定自定义 timestamp extractor 或者 Watermark Emission ，或者使用 [predefined ones]({{ site.baseurl }}/dev/event_timestamp_extractors.html)。你也可以通过以下方式将其传递给你的 consumer ：
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
你可以按照[此处]({{ site.baseurl }}/zh/dev/event_timestamps_watermarks.html)的说明指定自定义的  timestamp 抽取和 watermark 发送 ，或者使用 [内置的实现]({{ site.baseurl }}/dev/event_timestamp_extractors.html)。你也可以通过以下方式将其传递给你的 consumer ：
```

File: docs/dev/connectors/kafka.zh.md [hunk 16]
```diff
@@ -448,61 +383,36 @@ val stream = env.addSource(myConsumer)
...
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
 Properties properties = new Properties();
 properties.setProperty("bootstrap.servers", "localhost:9092");
-// only required for Kafka 0.8
+// 仅限于 Kafka 0.8 使用
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
// 仅 Kafka 0.8 需要
```

File: docs/dev/connectors/kafka.zh.md [hunk 17]
```diff
@@ -511,15 +421,15 @@ FlinkKafkaConsumer08<String> myConsumer =
...
+  .addSource(myConsumer)
+  .print();
 {% endhighlight %}
 </div>
 <div data-lang="scala" markdown="1">
 {% highlight scala %}
 val properties = new Properties()
 properties.setProperty("bootstrap.servers", "localhost:9092")
-// only required for Kafka 0.8
+// 仅限于 Kafka 0.8 使用
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
// 仅 Kafka 0.8 需要
```

File: docs/dev/connectors/kafka.zh.md [hunk 18]
```diff
@@ -532,41 +442,28 @@ stream = env
...
-
-**Note**: If a watermark assigner depends on records read from Kafka to advance its watermarks
-(which is commonly the case), all topics and partitions need to have a continuous stream of records.
-Otherwise, the watermarks of the whole application cannot advance and all time-based operations,
-such as time windows or functions with timers, cannot make progress. A single idle Kafka partition causes this behavior.
-A Flink improvement is planned to prevent this from happening
-(see [FLINK-5479: Per-partition watermarks in FlinkKafkaConsumer should consider idle partitions](
-https://issues.apache.org/jira/browse/FLINK-5479)).
-In the meanwhile, a possible workaround is to send *heartbeat messages* to all consumed partitions that advance the watermarks of idle partitions.
+在内部，每个 Kafka 分区执行一个 assigner 实例。当指定了这样的 assigner 时，对于从 Kafka 读取的每条消息，调用 `extractTimestamp(T element, long previousElementTimestamp)` 来为记录分配时间戳，并为 `Watermark getCurrentWatermark()` （定期）或 `Watermark checkAndGetNextWatermark(T lastElement, long extractedTimestamp)` （用于标点符号）以确定是否应该发出新的 watermark 以及使用哪个时间戳。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
在内部，每个 Kafka 分区执行一个 assigner 实例。当指定了这样的 assigner 时，对于从 Kafka 读取的每条消息，调用 `extractTimestamp(T element, long previousElementTimestamp)` 来为记录分配时间戳，并为 `Watermark getCurrentWatermark()` （定期形式）或 `Watermark checkAndGetNextWatermark(T lastElement, long extractedTimestamp)` （打点形式）以确定是否应该发出新的 watermark 以及使用哪个时间戳。
```

File: docs/dev/connectors/kafka.zh.md [hunk 19]
```diff
@@ -532,41 +442,28 @@ stream = env
...
+
+**请注意**： 如果水位线 assigner 依赖于从 Kafka 读取的消息来上涨其水位(通常就是这种情况)，那么所有主题和分区都需要有连续的消息流。否则，整个应用程序的水位将无法上涨，所有基于时间的算子(例如时间窗口或带有计时器的函数)也无法运行。单个的 Kafka 分区也会导致这种反应。这是一个已在计划中的 Flink improvement ，目的是为了防止这种情况发生（请见[FLINK-5479: Per-partition watermarks in FlinkKafkaConsumer should consider idle partitions](https://issues.apache.org/jira/browse/FLINK-5479)）。同时，可能的解决方法是将*心跳消息*发送到所有 consumer 的分区里，从而上涨空闲分区的水位。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
**请注意**： 如果 watermark assigner 依赖于从 Kafka 读取的消息来上涨其 watermark (通常就是这种情况)，那么所有主题和分区都需要有连续的消息流。否则，整个应用程序的 watermark 将无法上涨，所有基于时间的算子(例如时间窗口或带有计时器的函数)也无法运行。单个的 Kafka 分区也会导致这种反应。这是一个已在计划中的 Flink 改进，目的是为了防止这种情况发生（请见[FLINK-5479: Per-partition watermarks in FlinkKafkaConsumer should consider idle partitions](https://issues.apache.org/jira/browse/FLINK-5479)）。同时，可能的解决方法是将*心跳消息*发送到所有 consumer 的分区里，从而上涨空闲分区的 watermark。
```

我更倾向于对 watermark 保持同一种翻译，或者都不翻译。

File: docs/dev/connectors/kafka.zh.md [hunk 20]
```diff
@@ -532,41 +442,28 @@ stream = env
...
 
 ## Kafka Producer
 
-Flink’s Kafka Producer is called `FlinkKafkaProducer011` (or `010` for Kafka 0.10.0.x versions, etc. or just `FlinkKafkaProducer` for Kafka >= 1.0.0 versions).
-It allows writing a stream of records to one or more Kafka topics.
+Flink Kafka Producer 被称为 `FlinkKafkaProducer011` （或者 `010` 只适合 Kafka 0.10.0.x 的版本使用 ，或 `FlinkKafkaProducer` 只适用于 Kafka> = 1.0.0 的版本使用）。它允许将消息流写入一个或多个 Kafka topics 。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
Flink Kafka Producer 被称为 `FlinkKafkaProducer011` （或适用于 Kafka 0.10.0.x 版本的 `FlinkKafkaProducer010`，或适用于 Kafka >= 1.0.0 版本的 `FlinkKafkaProducer`）。它允许将消息流写入一个或多个 Kafka topics 。
```

File: docs/dev/connectors/kafka.zh.md [hunk 21]
```diff
@@ -532,41 +442,28 @@ stream = env
...
 
-Example:
+举例：
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
示例：
```

File: docs/dev/connectors/kafka.zh.md [hunk 22]
```diff
@@ -577,171 +474,101 @@ stream.addSink(myProducer);
...
+
+除了启用 Flink 的 checkpointing ，还可以通过将适当的 `semantic` 参数传递给 `FlinkKafkaProducer011` （适用于 Kafka >= 1.0.0 版本的 `FlinkKafkaProducer` ）来选择三种不同的操作模式：
+
+ * `Semantic.NONE`：Flink 不会有任何语义的保证，产生的记录可能会丢失或重复。
+ * `Semantic.AT_LEAST_ONCE` （默认设置）： 类似 `FlinkKafkaProducer010` 中的 `setFlushOnCheckpoint(true)` ，这可以保证不会丢失任何记录（虽然记录可能会重复）。
+ * `Semantic.EXACTLY_ONCE` ：使用 Kafka 事务提供精准一次的语义。无论何时，在使用事务写入 Kafka 时，都要记得为所有消费 Kafka 消息的应用程序设置所需的 `isolation.level`（ `read_committed` 或 `read_uncommitted`  - 后者是默认值）。
+
+##### 注意事项
+
+`Semantic.EXACTLY_ONCE` 模式依赖于，在从上述 checkpoint 恢复之后提交在获取 checkpoint 之前启动的事务的能力。如果 Flink 应用程序崩溃和完成重新启动之间的时间大于 Kafka 的事务超时，则会丢失数据（ Kafka 将自动中止超出超时时间的事务）。记住这一点，请根据预期的停机时间适当地配置事务超时时间。
```
Comment (wuchong) [inline docs/dev/connectors/kafka.zh.md]:
```suggestion
`Semantic.EXACTLY_ONCE` 模式依赖于事务提交的能力。事务提交发生于触发 checkpoint 之前，以及从 checkpoint 恢复之后。如果从 Flink 应用程序奔溃到完全重启的时间超过了 Kafka 的事务超时时间，那么将会有数据丢失（Kafka 会自动丢弃超出超时时间的事务）。考虑到这一点，请根据预期的宕机时间来合理地配置事务超时时间。
```

Comment (highfei2011) [reply docs/dev/connectors/kafka.zh.md]: Hi, Thanks your reviews ,  Here you may need to trouble you to modify the "奔溃" --> "崩溃".

File: docs/dev/connectors/kafka.zh.md [hunk 23]
```diff
@@ -107,40 +100,36 @@ Then, import the connector in your maven project:
 </dependency>
 {% endhighlight %}
 
-Note that the streaming connectors are currently not part of the binary distribution.
-See how to link with them for cluster execution [here]({{ site.baseurl}}/dev/projectsetup/dependencies.html).
+请注意：目前流连接器还不是二进制分发的一部分。
+[在此处]({{ site.baseurl}}/dev/projectsetup/dependencies.html)可以了解到如何链接它们以实现在集群中执行。
```
[the remark this answers is not in the document]
Comment (highfei2011) [reply docs/dev/connectors/kafka.zh.md]: Okay thank you! I will pay attention to this point in my future contributions.

File: docs/dev/connectors/kafka.zh.md (hunk 19 above)
Comment (highfei2011) [reply docs/dev/connectors/kafka.zh.md]: 我也觉得 “watermark” 应该是一个专用名词，不应该翻译其字面的意思。

File: docs/dev/connectors/kafka.zh.md (hunk 22 above)
Comment (wuchong) [reply docs/dev/connectors/kafka.zh.md]: You are right. It should be "崩溃".  ;)

Comment (highfei2011) [reply docs/dev/connectors/kafka.zh.md]: Thanks ,I have modified it.

File: docs/dev/connectors/kafka.zh.md [hunk 24]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Apache Kafka Connector"
+title: "Apache Kafka 连接器"
```
Comment (TisonKun) [inline docs/dev/connectors/kafka.zh.md]: @wuchong  shall we translate proper nouns such as "connector"? Do we have a general translation with consensus or at least listed out in Glossary? IMHO there is no need to translate "connector".

Comment (wuchong) [reply docs/dev/connectors/kafka.zh.md]: Yes. We have a consensus on connector.
https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications

**Comment by highfei2011** [comment]
Open a new PR.

**Pull request closed without merging**